kpops/utils/environment.py

```diff
@@ -0,0 +1,34 @@
+import os
+import platform
+from collections import UserDict
+from typing import Callable
+
+
+class Environment(UserDict):
+    def __init__(self, mapping=None, /, **kwargs) -> None:
+
+        transformation = self._get_transformation()
+        if mapping is not None:
+            mapping = {transformation(key): value for key, value in mapping.items()}
+        else:
+            mapping = {}
+        if kwargs:
+            mapping.update(
+                {transformation(key): value for key, value in kwargs.items()}
+            )
+        super().__init__(mapping)
+
+    def _key_camel_case_transform(self, key: str) -> str:
+        return str(key).lower()
+
+    def _key_identity_transform(self, key: str) -> str:
+        return str(key)
```

Comment:
Well it depends... right now could be redundant but a dict key can be anything. So, we use it for os.environ and that's why we can say for sure it is a string, but it can multiple things. I can change it if you want.

Reply:
imo it doesn't have to be general purpose, as we only need it for environment. Also the parameter type annotation says it's always a `str`